src/ape_plugins/_cli.py

```diff
@@ -181,8 +217,22 @@ def install(cli_ctx, skip_confirmation):
             any_install_failed = True
         # check for installed check the config.yaml
         elif installed_plugin and available_plugin:
-            cli_ctx.logger.warning(f"Plugin '{module_name}' is trusted and already installed.")
-            pass
+            cli_ctx.logger.info(f"Plugin '{module_name}' is trusted and already installed.")
+            if upgrade:
+                cli_ctx.logger.info(f"Updating {module_name} plugin.")
+                args.append("--upgrade")
+                args.append(module_name)
+                result = subprocess.call(args)
+                if result == 0 and is_plugin_installed(module_name):
+                    cli_ctx.logger.success(f"Plugin '{module_name}' has been upgraded.")
+                else:
+                    cli_ctx.logger.error(f"Failed to add '{module_name}'.")
+                    sys.exit(1)
```

Comment:
Hmmm... Do we want to fail the whole command if a single plugin fails to install properly? Were we doing this before?

Reply:
I think we can remove the `sys.exit(1)` here do what we do below where we track if any of them failed and then exit with a status of `1` at the very end

Reply:
I thin you are right and we should try and install as many as we can and just show error loggers and sys exit at the end
I removed the sys exit and place a flag so it can continue run and just exit with 1 at the end